src/main/kotlin/net/thoughtmachine/please/plugin/runconfiguration/PleaseLineMarkerProvider.kt

```diff
@@ -54,19 +55,37 @@ object PleaseLineMarkerProvider : RunLineMarkerContributor() {
                 val target = "//${file.getPleasePackage()}:${expr.stringValue}"
                 return Info(
                     AllIcons.Actions.Execute,
-                    { "run $target" },
-                    PleaseAction(element.project, target, "run", DefaultRunExecutor.getRunExecutorInstance(), newRunConfig(element.project, target)),
-                    PleaseAction(element.project, target, "run", DefaultDebugExecutor.getDebugExecutorInstance(), newRunConfig(element.project, target)),
-                    PleaseAction(element.project, target, "test", DefaultRunExecutor.getRunExecutorInstance(), newTestConfig(element.project, target)),
-                    PleaseAction(element.project, target, "test", DefaultDebugExecutor.getDebugExecutorInstance(), newTestConfig(element.project, target)),
-                    PleaseAction(element.project, target, "build", PleaseBuildExecutor, newBuildConfig(element.project, target))
-                )
+                    filterGoActions(element, target).toTypedArray(),
+                ) { "run $target" }
             }
         }
 
         return null
     }
 
+    private fun filterGoActions(element: PsiElement, target: String): List<AnAction> {
+        val actions = mutableListOf(PleaseAction(element.project, target, "build",
+            PleaseBuildExecutor, newBuildConfig(element.project, target)))
+
+        // TODO(jscott) we should find a way of making this more generic for other languages
```

Comment:
Yeah, ideally we'd be using `plz query --json` to figure out if it's binary, and if it's a test. We could pass this information to the debug providers and see if any of them support debugging the rule.